src/java.base/share/classes/java/net/DatagramSocket.java

```diff
@@ -1398,14 +1391,11 @@ private static SocketException toSocketException(IOException e) {
         boolean initialized = false;
         try {
             DatagramSocketImplFactory factory = DatagramSocket.factory;
-            if (USE_PLAINDATAGRAMSOCKET || factory != null) {
+            if (factory != null) {
                 // create legacy DatagramSocket delegate
-                DatagramSocketImpl impl;
-                if (factory != null) {
-                    impl = factory.createDatagramSocketImpl();
-                } else {
-                    impl = DefaultDatagramSocketImplFactory.createDatagramSocketImpl(multicast);
-                }
+                DatagramSocketImpl impl = factory.createDatagramSocketImpl();
+                Objects.requireNonNull(impl,
+                        "Implementation returned by installed DatagramSocketFactoryImpl is null");
```

Comment:
I think you mean "DatagramSocketFactoryImpl" rather than "DatagramSocketImplFactory" here.

Reply:
Yes - good catch! I've been bitten by this before too  :-) 
https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/net/DatagramSocketImplFactory.html

Reply:
Well spotted. Thanks Alan. Change made in commit 686b436